### ISSUE TITLE
Refactor getmodel Result struct to hold doubles instead of floats?

### ISSUE THREAD

**Issue by cdlhub** [OasisLMF/ktools#10]
Because construction of `Result` cast a double like in:

```cpp
double cumulative_prob = 0;
//...
results[result_index++] = Result(static_cast<float>(cumulative_prob), _mean_damage_bins[damage_bin_index]);
```

Would not it be better to have `Result` defined with doubles, which is also way better for numeric computation?:

```cpp
struct Result {
  Result(double damage, double prob);

  double prob;
  double damage;
};
```

**Comment by bmatharu** [comment]
Other than the higher precision computation, doubles are not always necessarily faster than single precision calculations - We made a design decision long time ago that our streams will always use 4 byte floats and 4 byte integers - This was to limit the volume of data fed through the system. Some of the larger models are known to stream a terabyte of data - we found that even for simple models streaming floats vs doubles could reduce the volume of data by a third.
Although this is our reasoning we appreciate others may not see things the same way - which is why I think we will move towards making the ability to change the data types easier. A user in Australia did convert his copy to doubles for his needs.
